pkg/nimble/scanner/nimble_scanner.c

```diff
@@ -103,13 +103,16 @@ int nimble_scanner_start(void)
         const struct ble_gap_ext_disc_params *coded =
             (_scan_flags & NIMBLE_SCANNER_PHY_CODED) ? &_scan_params : NULL;
 
+        int32_t dur = (_scan_duration == BLE_HS_FOREVER) ? 0
+                                                         : _scan_duration / 10;
+
         int res = ble_gap_ext_disc(nimble_riot_own_addr_type,
-                                   _scan_duration / 10, 0, dups,
+                                   _scan_duration, 0, dups,
                                    BLE_HCI_SCAN_FILT_NO_WL, limited,
                                    uncoded, coded,
                                    _on_scan_evt, NULL);
 #else
-        int res = ble_gap_disc(nimble_riot_own_addr_type, 0,
+        int res = ble_gap_disc(nimble_riot_own_addr_type, _scan_duration,
```

Comment:
```suggestion
        int res = ble_gap_disc(nimble_riot_own_addr_type, dur,
```

Reply:
Nope, the non-extended discovery has the timeout given in `ms` and uses `BLE_HS_FOREVER` as value for no timeout, so we use `_scan_duration` here directly.